### ISSUE TITLE
1.16.5 v1.3.3 - How to Disenchant?

### ISSUE THREAD

**Issue by trentnbauer** [CplPibald/PiTweaks#22]
G'day mate,

Feeling pretty stupid, but unsure how to get this to work.

I've enabled disenchant in the config and rebooted the server. I've tried putting my helmet in the anvil with a normal book, I've tried in the crafting table and in the upgrade station without any luck.
Any suggestions?

Server is running Better Minecraft v57, if that matters.

Contents of my piteaks.toml file

> #Anvil Tweaks
> [anvil]
> 	#Removes additional cost from repairing/enchanting items in an anvil multiple times
> 	noRepairCost = false
> 	#Allow books to be combined over the enchantment max level
> 	allowOverlevelBooks = false
> 	#Always allow enchantments from books to be applied in an anvil, regardless of item type or other enchantments
> 	alwaysAllowBooks = false
> 	#Enables disenchanting items by combining with a vanilla book
> 	disenchant = true
> 	#Chance for an anvil to be damaged on use (default 0.12)
> 	#Range: 0.0 ~ 1.0
> 	breakChance = 0.12
> 
> #Vanilla Machine Speed Tweaks
> [machines]
> 	#Vanilla furnaces are faster.
> 	fastFurnace = false
> 	#Vanilla blast furnaces are faster.
> 	fastBlastFurnace = false
> 	#Vanilla smokers are faster.
> 	fastSmoker = false
> 	#Vanilla campfires cook faster.
> 	fastCampfire = true
> 	#Cook speed multiplier for furnace, blast furnace, and smoker. Time per item is (10/multiplier) seconds. Valid values (2,4,5,8,10,20,25,40,50)
> 	#Range: 2 ~ 50
> 	multiplier = 2
> 	#Brewing stands are faster.
> 	fastBrewing = false
> 	#Number of ticks to brew a potion.  Vanilla = 400.
> 	#Range: 2 ~ 400
> 	brewTicks = 400
> 
> #Entity Tweaks
> [entity]
> 	#Mobs can only spawn blocks with a full-cube collision box.  Prevents spawning on partial blocks like top-slabs, upside-down-stairs, and hoppers.
> 	spawnRequiresFullCube = false
> 	#Set to false to disable phantom spawns.
> 	allowPhantoms = true
> 	#Removes 0.5 sec invulnerability time after an entity gets hit.
> 	noInvulnerabilityTicks = false
> 	#Crouch while flying to immediately end elytra flight.
> 	crouchStopsFlight = false
> 
> #Hunger Tweaks
> [food]
> 	#Remove food cap that limits food value to 20. Useful with modded foods
> 	uncapFood = true
> 	#Remove saturation cap that limits saturation to current food level
> 	uncapSaturation = true
> 
> #Plant Tweaks
> [plant]
> 	#Leaves decay immediately when their supporting tree is removed
> 	instantLeafDecay = false

**Comment by CplPibald** [comment]
Yep, that's a logic bug.  The Anvil tweaks module only gets loaded if `allowOverlevelBooks` or `alwaysAllowBooks` is turned on.  I need to make sure to load it when `disenchant` is on too.

Workaround: Turn one of those other two settings on (and restart server).

After that, you should be able to disenchant in an anvil.  Put an enchanted item in the left slot and a vanilla book in the right.

**Comment by trentnbauer** [comment]
Thanks for getting back to me. I've enabled 'allowoverlevelbooks' but I'm a bit nervous that someone might create a level 20 sharpness sword.
Is this a Minecraft issue, or something you can sort out with a patch?
